app/services/request_id_validator.rb

```diff
@@ -0,0 +1,37 @@
+class RequestIdValidator
+  include ActiveModel::Model
+  include ActionView::Helpers::TranslationHelper
+
+  validate :validate_request_id
+
+  def initialize(request_id)
+    @request_id = request_id
+  end
+
+  def submit
+    success = valid?
+
+    FormResponse.new(
+      success: success,
+      errors: errors,
+      extra: extra_attributes
+    )
+  end
+
+  private
+
+  attr_reader :request_id
+
+  def validate_request_id
+    return if request_id.blank?
+    return if ServiceProviderRequest.where(uuid: request_id).exists?
+
+    errors.add(:request_id, :invalid)
+  end
+
+  def extra_attributes
+    {
+      request_id: request_id,
```

Comment:
I thought of a better idea. How about tracking the request_id automatically everywhere by changing `request_attributes` in `analytics.rb` to this:
```ruby
def request_attributes
  {
    user_ip: request.remote_ip,
    user_agent: request.user_agent,
    host: request.host,
    request_id: request.parameters.fetch(:request_id, nil),
  }
end
```